samples/runtime/qpic-simulator/simulator/Processor.cs

```diff
@@ -0,0 +1,186 @@
+            AddCommand(format, qubits.ToArray<object>());
+        }
+
+        public override void Reset(Qubit qubit) {
+            AddCommand("{0} OUT {{0}}", qubit);
+        }
+        #endregion
+
+        #region Classical control
+        // This is a custom Result class which can hold the qubit that is being
+        // measured.  In this simulator we are not interested in the simulation
+        // value, but need to keep track of qubits that were being measured to
+        // support classical control.
+        class DelayedResult : Result {
+            public Qubit Qubit { get; set; }
+
+            // Since we are not interested in the outcome of a measurement, we
+            // simply return One.  This method needs to be implemented to avoid
+            // an exception from being thrown, but the value will not be used by
+            // the simulator.
+            public override ResultValue GetValue() {
+                return ResultValue.One;
+            }
+        }
+
+        // A measurement is used to keep track of the measured qubit, and adds a
+        // qpic measurement command.
+        public override Result M(Qubit qubit) {
+            AddCommand("{0} M", qubit);
+            return new DelayedResult { Qubit = qubit };
+        }
+
+        // When starting a conditional statement, the measured qubit is obtained
+        // from the measurementResult argument.  The inversion flag is
+        // initialized with respect to the result value.  If it's `Zero`,
+        // operations in the `then` clause are negatively controlled, and
+        // operations in the `else` clause are positively controlled.  If it's
+        // `One`, it's the other way around.  Both variables are inserted into
+        // the dictionary for classically controlled qubits.  The qubit's id is
+        // used as identifier to link successive methods to the same conditinal
+        // statement.
+        public override long StartConditionalStatement(Result measurementResult, Result resultValue) {
+            var qubit = ((DelayedResult)measurementResult).Qubit;
+            var invert = resultValue == Result.Zero;
+            classicallyControlledQubits.Add(qubit.Id, (qubit, invert));
+            return qubit.Id;
+        }
+
+        // No action needs to be taken in the `RunThenClause` method.  It
+        // returns true to ensure that the corresponding operations in the
+        // `then` clause are executed.
+        public override bool RunThenClause(long statement) => true;
+
+        // In the `RunElseClause` method the invert flag of the corresponding
+        // dictionary entry is toggled.  The method also returns true to ensure
+        // that the corresponding operations in the `else` clause are executed.
+        public override bool RunElseClause(long statement) {
```

Comment:
It may be helpful to have a comment explaining how the `statement` parameter is to be interpreted.

Reply:
Will add a clarification in the next commit.